R/get-datasets.R

```diff
@@ -44,18 +44,27 @@ datasets <- function (x=getAPIRoot()) {
 #' the primary dataset catalog for the user will be used.
 #' @param refresh logical: should the function check the Crunch API for new
 #' datasets? Default is FALSE.
+#' @param shiny logical: launch a shiny gadget to help select the right dataset. The
+#' gadget will return a valid `loadDataset()` call which loads the selected dataset.
 #' @return Character vector of dataset names, each of which would be a valid
 #' input for [loadDataset()]
 #' @export
 listDatasets <- function (kind=c("active", "all", "archived"), project=NULL,
-                          refresh=FALSE) {
-    dscat <- selectDatasetCatalog(kind, project, refresh)
-    return(names(dscat))
+    refresh=FALSE, shiny = FALSE) {
+    if (shiny) {
+        if (rstudioapi::isAvailable("0.99.878")) {
```

Comment:
Since `rstudioapi` is under Suggests, you can't assume that it is installed here. https://github.com/Crunch-io/rcrunch/blob/master/R/auth.R#L86 is one way to check (without loading it); I think you can use `require` or something along those lines too, and here's how HH Hadley does it in httr: https://github.com/r-lib/httr/blob/master/R/utils.r#L34-L42